src/main/java/com/google/sps/servlets/CheckPermissionsServlet.java

```diff
@@ -16,11 +16,77 @@
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
 
+// import com.google.api.client.googleapis.auth.oauth2.GoogleCredential;
+// import com.google.api.client.googleapis.javanet.GoogleNetHttpTransport;
+// import com.google.api.client.http.HttpTransport;
+// import com.google.api.client.json.JsonFactory;
+// import com.google.api.client.json.jackson2.JacksonFactory;
+// import com.google.api.services.iam.v1.iam;
+// import com.google.api.services.iam.v1.model.ListServiceAccountsResponse;
+// import com.google.api.services.iam.v1.model.ServiceAccount;
+// import java.io.IOException;
+// import java.security.GeneralSecurityException;
+// import java.util.Arrays;
+
+import com.google.api.client.googleapis.javanet.GoogleNetHttpTransport;
+import com.google.api.client.json.jackson2.JacksonFactory;
+import com.google.api.services.iam.v1.Iam;
+import com.google.api.services.iam.v1.IamScopes;
+import com.google.api.services.iam.v1.model.ListServiceAccountsResponse;
+import com.google.api.services.iam.v1.model.ServiceAccount;
+import com.google.auth.http.HttpCredentialsAdapter;
+import com.google.auth.oauth2.GoogleCredentials;
+import java.io.IOException;
+import java.security.GeneralSecurityException;
+import java.util.Collections;
+import java.util.List;
+import com.google.gson.Gson;
+
 @WebServlet("/check-permissions")
 public class CheckPermissionsServlet extends HttpServlet {
   
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-      System.out.println("It works");
+    String projectId = request.getParameter("projID");
+
+  // Lists all service accounts for the current project.
+  //public static void listServiceAccounts(String projectId) {
+
+    Iam service = null;
+    try {
+      service = initService();
+    } catch (IOException | GeneralSecurityException e) {
+      System.out.println("Unable to initialize service: \n" + e.toString());
+      return;
+    }
+
+    try {
+      ListServiceAccountsResponse accountResponse =
+          service.projects().serviceAccounts().list("projects/" + projectId).execute();
+      List<ServiceAccount> serviceAccounts = accountResponse.getAccounts();
+
+      response.setContentType("application/json;");
+      Gson gson = new Gson();
+      response.getWriter().println(gson.toJson(serviceAccounts));
+
+    } catch (IOException e) {
+      System.out.println("Unable to list service accounts: \n" + e.toString());
+    }
+  }
+
+  private static Iam initService() throws GeneralSecurityException, IOException {
+    // uses the Application Default Credentials strategy for authentication.
+    GoogleCredentials credential =
+        GoogleCredentials.getApplicationDefault()
+            .createScoped(Collections.singleton(IamScopes.CLOUD_PLATFORM));
```

Comment:
I think this command uses your default service account instead of the one provided by the user.

Reply:
After checking with @tblanshard, I think we will need to ask the customer to upload a JSON with the key of the service account, and use that to authenticate as the service account. There is a code snippet at https://cloud.google.com/docs/authentication/production#manually